src/planner/planner.cpp

```diff
@@ -0,0 +1,157 @@
+#include "planner/planner.h"
+#include "binder/expressions/bound_agg_call.h"
+#include "binder/expressions/bound_column_ref.h"
+#include "binder/table_ref/bound_base_table_ref.h"
+#include "execution/expressions/aggregate_value_expression.h"
+#include "execution/expressions/column_value_expression.h"
+#include "execution/plans/aggregation_plan.h"
+#include "execution/plans/seq_scan_plan.h"
+
+namespace bustub {
+
+void Planner::PlanQuery(const BoundStatement &statement) {
+  switch (statement.type_) {
+    case StatementType::SELECT_STATEMENT: {
+      plan_ = PlanSelect(dynamic_cast<const SelectStatement &>(statement));
+      return;
+    }
+    default:
+      throw Exception("the statement is not supported in planner yet");
```

Comment:
Can you use fmt to print the statement type in the error message? If it is printable.